src/graphs/Breadth_First_Search.re

```diff
@@ -0,0 +1,65 @@
+type node = {
+    id: string,
+    neighbours: list(node),
+};
+
+type pair = (string, list(string));
+
+let approx_num_nodes = 5;
+let level: Hashtbl.t(string, int) = Hashtbl.create(approx_num_nodes);
+
+let toHashTbl = pair_list => {
```

Comment:
I don't think you need this helper function.
Since `node` is exposed you can expect the users to give you a list of nodes.
If not that, you can expose this function and hide the implementation of `node` and return both a `parent` hash and `level` hash.

Reply:
Agree with `node` not needing to be exposed. Agree with `parent` hash table addition.

My reasoning for this was that it would be more efficient if `adj_list` was a hash table rather than a list (faster lookups). But I also thought it would be cumbersome for the user to have to create a hash table as their `adj_list`, so I let them pass in an `adj_list` in the form of an actual List which then internally gets converted into a hash table.

My understanding is that in the 2nd option you gave, you want me to expose this function and let the user convert `adj_list` to a hash table themselves instead of handling that internally, am I correct?

Reply:
Let users pass in a list of nodes instead of pairs.
Then this function should both validate the list (what happens when a node in `adj_list` doesn't exist) and then turns it into a `Hashtbl`.